recipe/run_conda_forge_build_setup_linux

```diff
@@ -12,7 +12,7 @@ conda config --set add_pip_as_python_dependency false
 
 conda update -n root --yes --quiet conda conda-env conda-build
 conda install -n root --yes --quiet jinja2 anaconda-client
-conda install -n root --yes --quiet conda-build=2
+conda install -n root --yes --quiet conda-build=3
```

Comment:
I think this is one of those cases where, as part of the upgrade, we are going to need to temporarily support both `conda-build` 2 and 3. Otherwise we are going to get into a bit of a deadlock. Is it possible to soften this to use `conda-build` 2 and 3 for now (even if it is pretty hacky)?

Reply:
I don't think I understand how you would distinguish when to run 2, and when to run 3.  Maybe I just don't understand how conda-forge-build-setup is used.  I understood it to be something of a central setup mechanism, with each feedstock also free to add stuff instead of scripts here.

Reply:
Also, I'd like to understand use cases where 2 is necessary.  3 (especially 3.1, with the relaxed hashes) should be pretty much a drop-in replacement.  Can you explain use cases where 3.1 doesn't work and you need 2.1.18?

Reply:
@jakirkham can we please do something to move this forward?  What do you want to see here?

I think the python code changes below are compatible with conda-build 2 and 3.  We could set an environment variable for the conda-build version?  I guess that's something conda-smithy would do?

Reply:
I'd like to, but I see at least one challenge. Namely current feedstocks won't work with `conda-build` 3 without re-rendering them, correct? They are still dependent on environment variables. So if we keep this as `conda-build` 2 for now, we can certainly get out there. We will need to get a new `conda-smithy` out so that we can actually switch to `conda-build` 3 AFAICT. The latter I don't think I'll have time for before the break, but this part we certainly could try before the break with that change.

Reply:
No, current feedstocks will absolutely work with conda-build 3.  If they don't, that's a bug.  They won't take advantage of conda-build 3 for build matrix computation until we re-render them, but that's OK.

If we keep this as CB2, I don't understand how we switch CB3 on.  I hacked conda-smithy to add in an installation of CB3 after these scripts from conda-forge-build-setup had run.  Changing it here is the right place, as long as CB3 works in place of CB2 (which it absolutely should).

What tests would help prove this to you?

Reply:
Maybe I'm misunderstanding, but had the impression that environment variables would not work with `conda-build` 3.

Reply:
@msarahan, if current feedstocks will work with conda-build 3, I think we just need to fix `staged-recipes` so that it'll use conda-build 2 and then we can merge this.

Reply:
Do you want to put up a PR over at `staged-recipes` @isuruf?

Reply:
FWIW I think we only need the pinning for feedstock conversion. It shouldn't be needed when building recipes as `conda-build-all` should already support `conda-build` 3.

Reply:
environment variables such as CONDA_PY, CONDA_R, CONDA_NPY will absolutely work.  Those have always worked.  What was broken for some time was accessing attributes of conda-build's config class, like config.CONDA_PY.  Those have been restored, but deprecated.  conda-smithy and conda-build-all were the main users.

Reply:
Ok, thanks for the info.

Reply:
> ...I think we just need to fix `staged-recipes` so that it'll use conda-build 2 and then we can merge this.

Doing in PR ( https://github.com/conda-forge/staged-recipes/pull/4685 ).